Wrong chart size in Firefox browser #3683

kpaharelau opened the pull request:
use simple width() and height() functions in a firefox browser.

kpaharelau committed:
Wrong chart size in Firefox browser #3683

use simple width() and height() functions in a firefox browser.

scottdraves commented on core/src/main/web/outputdisplay/bko-plot/plotutils.js:
```diff
@@ -54,11 +54,24 @@
       };
     return {
+
+      isFirefox: function ()
```
these should be application-wide helper functions, not just here.
in fact i am surprised we don't have something like this already, please check.

scottdraves commented on core/src/main/web/outputdisplay/bko-plot/plotutils.js:
```diff
@@ -54,11 +54,24 @@
+      {
+        return typeof InstallTrigger !== 'undefined';
+      },
+      isSafari: function ()
```
should be formatted

```
function () {
 return ...
}
```

kpaharelau committed:
Wrong chart size in Firefox browser #3683

these should be application-wide helper functions, not just here.
in fact i am surprised we don't have something like this already, please check.

I moved functionality to the GLOBALS. I did not find something like this. I searched by words "firefox", 'chrome', "browser"

scottdraves commented on core/src/main/web/app/helpers/globals.js:
```diff
@@ -50,7 +50,22 @@
         AMBIANCE: 'ambiance'
-      }
+      },
+      //see http://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser
```
should go in helper.js, global.js is just for constants not for functions.

also, please only define firefox, safari, and chrome as we support only them.

kpaharelau committed:
Wrong chart size in Firefox browser #3683

should go in helper.js, global.js is just for constants not for functions.

also, please only define firefox, safari, and chrome as we support only them.

scottdraves commented on core/src/main/web/app/helpers/helper.js:
```diff
@@ -73,6 +73,14 @@
+      // Firefox 1.0+
+      isFirefox: typeof InstallTrigger !== 'undefined',
+      // At least Safari 3+: "[object HTMLElementConstructor]"
+      isSafary: Object.prototype.toString.call(window.HTMLElement).indexOf('Constructor') > 0,
```
Safari not Safary

kpaharelau committed:
Wrong chart size in Firefox browser #3683

Safari not Safary
